presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -274,14 +284,15 @@ private static String formatFragment(FunctionRegistry functionRegistry, StatsCal
         }
         builder.append(indentString(1)).append(format("Grouped Execution: %s\n", fragment.getStageExecutionStrategy().isAnyScanGroupedExecution()));
 
-        if (stageInfo.isPresent()) {
-            builder.append(textLogicalPlan(fragment.getRoot(), TypeProvider.copyOf(fragment.getSymbols()), Optional.of(fragment.getStageExecutionStrategy()), functionRegistry, statsCalculator, costCalculator, session, Optional.of(planNodeStats.get()), 1, verbose))
-                    .append("\n");
-        }
-        else {
-            builder.append(textLogicalPlan(fragment.getRoot(), TypeProvider.copyOf(fragment.getSymbols()), Optional.of(fragment.getStageExecutionStrategy()), functionRegistry, statsCalculator, costCalculator, session, Optional.empty(), 1, verbose))
-                    .append("\n");
-        }
+        TypeProvider typeProvider = TypeProvider.copyOf(allFragments.stream()
```

Comment:
@rschlussel : My understanding is we need `allFragments` as we want to create `sourceProvider` and `statsCalculator`. However, later you make stats and cost precomputed and store in `PlanFragment`: https://github.com/prestodb/presto/pull/11511

I am wondering do we still need to instantiate the `TypeProvider` with all fragments?